#17785: Added Swin_S trace pipeline

HariniMohan0102 opened the pull request:
### Ticket
[#17785](https://github.com/tenstorrent/tt-metal/issues/17785)

### Problem description
- To enable Swin_S trace pipeline

### What's changed
- Added Swin_S trace 2CQ pipeline
- E2E FPS: 5

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/16714723011) - CI Passed
- [x] [Model regression](https://github.com/tenstorrent/tt-metal/actions/runs/16714728833) - CI Passed
- [x] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/runs/16714725201) - CI Passed
- [x] [Frequent model & ttnn tests](https://github.com/tenstorrent/tt-metal/actions/runs/16714730852) - Swin_s Passed

HariniMohan0102 committed:
17785: Added Swin_S trace pipeline

HariniMohan0102 commented on models/experimental/swin_s/runner/performant_runner_infra.py:
```diff
@@ -0,0 +1,144 @@
+        return tt_inputs_host, sharded_mem_config_DRAM, input_mem_config
+
+    def run(self):
+        self.output_tensor = self.ttnn_swin_model(self.input_tensor)
```
The run function is called from performant runner file, hence the self.input_tensor is not defined in this file.

ctr-rsaravananTT commented on models/experimental/swin_s/runner/performant_runner_infra.py:
```diff
@@ -0,0 +1,145 @@
+        if is_wormhole_b0():
+            core_grid = ttnn.CoreGrid(y=8, x=8)
+        else:
+            exit("Unsupported device")
```
always raise exception and mention the reason for exception in the string

HariniMohan0102 replied:
Modified as suggested.

bbradelTT commented on models/experimental/swin_s/runner/performant_runner_infra.py:
```diff
@@ -0,0 +1,145 @@
+
+        n, c, h, w = torch_input_tensor.shape
+
+        padded_c = 16 if c == 3 else c  # If the channels < 16, pad the channels to 16 to run the Conv layer
```
There are values other than 3 that are < 16. It would be good to explain what is special about 3.

Also, this is all over the place. Might be worthwhile to have a common method.

HariniMohan0102 replied:
The input tensor shape for this model has channels as 3, hence mentioned explicitly. 
Will modify it generically.

HariniMohan0102 replied:
Modified code to be generic.

arginugaTT commented:
@HariniMohan0102 

Please check: `pytest models/experimental/swin_s/tests/perf/test_perf.py`

`PASSED models/experimental/swin_s/tests/perf/test_perf.py::test_swin_s[input_shape0-117-0.23-device_params0]
FAILED models/experimental/swin_s/tests/perf/test_perf.py::test_perf_device_bare_metal_swin_s[1-Swin_s-5.3] - subprocess.CalledProcessError: Command '['python3 -m tracy -p -r -o /home/ubuntu/tt-metal/generated/profiler/Swin_s --check-exit-code  -t 5000 -m pytest tests/ttnn/integration_tests/swin_s/test_ttnn_swin_transformer.py::test_swin_s_transfor...`

HariniMohan0102 commented:
@arginugaTT 

- Increased the iterations to 50 in e2e performant file.
- Removed the print of output in performant runner file already.
- Rebased the branch to latest main and here are the recent CI links:
  - [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml): [Failed, again Triggered CI in progress](https://github.com/tenstorrent/tt-metal/actions/runs/16418421495)
  - [x] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml): [Queued](https://github.com/tenstorrent/tt-metal/actions/runs/16413504568/job/46374515789)
  - [x] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml): [Swin_S Passed](https://github.com/tenstorrent/tt-metal/actions/runs/16413510093/job/46374638970)
  - [x] [Integeration tests](https://github.com/tenstorrent/tt-metal/actions/workflows/fast-dispatch-full-regressions-and-models.yaml): [Swin_S Passed](https://github.com/tenstorrent/tt-metal/actions/runs/16413516202)

- Can you please share the links where the swin_s perf has failed?

HariniMohan0102 commented on tests/scripts/run_performance.sh:
```diff
@@ -31,6 +31,8 @@ run_perf_models_other() {
 
         env WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest -n auto models/demos/yolov7/tests/ -m $test_marker
 
+        # env WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest -n auto models/experimental/swin_s/tests -m $test_marker
```
Since the run_performance.sh file is removed, added the test to perf-models-impl.yaml.

HariniMohan0102 commented:
Latest CI links:
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml): [CI Passed](https://github.com/tenstorrent/tt-metal/actions/runs/16541818220)
- [x] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml): [Swin_S Passed](https://github.com/tenstorrent/tt-metal/actions/runs/16541819817/job/46784192802)
- [x] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml): [CI Passed](https://github.com/tenstorrent/tt-metal/actions/runs/16541822298/job/46784208817)
- [x] [Integeration tests](https://github.com/tenstorrent/tt-metal/actions/workflows/fast-dispatch-full-regressions-and-models.yaml): [Swin_S Passed](https://github.com/tenstorrent/tt-metal/actions/runs/16541823818/job/46784200184)

saichandax commented:
New CIs triggered. Updated in the PR description

saichandax committed:
#0: Fix imports

saichandax committed:
Merge branch 'main' into harini/swin_trace
